bin/main.ml

```diff
@@ -12,23 +12,35 @@ let file =
   let doc = "File to check links." in
   Arg.(required & pos 1 (some file) None & info [] ~docv:"FILE" ~doc)
 
-let olinkcheck format file =
+let annotate_in_file =
+  let doc = "Annotate broken links in the file" in
+  Arg.(value & flag & info [ "annotate-in-file" ] ~doc)
+
+let olinkcheck annotate_in_file format file =
   match format with
   | Markdown ->
       `Ok
-        Markdown.(
-          Utils.pretty_print_link_status_from_file ".md" from_string
-            extract_links file)
+        (if annotate_in_file then
+           Markdown.(
+             Utils.annotate_in_file ".md" from_string extract_links file)
+         else
+           Markdown.(
+             Utils.pretty_print_link_status_from_file ".md" from_string
+               extract_links file))
   | Plaintext ->
       `Ok
-        Plaintext.(
-          Utils.pretty_print_link_status_from_file ".txt" from_string
-            extract_links file)
+        (if annotate_in_file then
+           Plaintext.(
+             Utils.annotate_in_file ".txt" from_string extract_links file)
+         else
+           Plaintext.(
+             Utils.pretty_print_link_status_from_file ".txt" from_string
+               extract_links file))
```

Comment:
Looks a bit repetitive to me. Maybe we can do the check for `annotate_in_file` before the match statement?

Like this, maybe? (untested)

```ocaml
let f =
  if annotate_in_file then Utils.annotate_in_file
  else Utils.pretty_print_link_status_from_file
in

match format with
| Markdown -> `Ok Markdown.(f ".md" from_string extract_links file)
| Plaintext -> `Ok Plaintext.(f ".txt" from_string extract_links file)
```

Reply:
I've moved the `annotate_in_str` function to `lib` to handle some format-specific cases.